src/sentry/options/types.py

```diff
@@ -0,0 +1,173 @@
+"""
+sentry.options.types
+~~~~~~~~~~~~~~~~~~~~
+
+:copyright: (c) 2010-2015 by the Sentry Team, see AUTHORS for more details.
+:license: BSD, see LICENSE for more details.
+"""
+from __future__ import absolute_import, print_function
+
+from sentry.utils.yaml import safe_load
+from yaml.parser import ParserError
+from yaml.scanner import ScannerError
+
+__all__ = (
+    'InvalidTypeError',
+    'Any', 'Bool', 'Int', 'Float', 'String', 'Dict', 'Sequence',
+)
+
+
+class InvalidTypeError(TypeError):
+    pass
+
+
+class OptionType(object):
+    """Base OptionType that provides type coersion"""
+    name = ''
+    # Default value to be returned when initializing
+    default = None
+    # Types that do not need to be coerced
+    expected_types = (object,)
+    # Types that are acceptable for coersion
+    compatible_types = (basestring,)
+
+    def __call__(self, value=None):
+        if value is None:
+            return self._default()
+        if self.test(value):
+            return value
+        if isinstance(value, self.compatible_types):
+            rv = self.convert(value)
+            # Make sure convert was able to do the right thing
+            # and give us the type we were expecting
+            if self.test(rv):
+                return rv
```

Comment:
To be clear, I think the `test` on line 43 is still necessary — I just think that the conversion methods should be raising an exception if they know that they are failing instead of returning an error code (in this case, `None`.) I think the indirection of catching errors and returning `None` just to throw another different exception with less context is *more* confusing, personally.

This only works (for now at least) since `None` isn't a valid option — which I still think is weird, and is likely influencing my viewpoint here. If we ever need a `None` option, this entire API will have to change.